FreeRTOS/Test/CMock/smp/granular_lock/covg_granular_lock_tasks_utest.c

```diff
@@ -0,0 +1,616 @@
+{
+    return numFailures;
+}
+
+/* ==============================  Test Cases tasks ============================== */
+
+/**
+ * @brief xTaskRemoveFromEventList - Remove a task from event list.
+ *
+ * The task event list item will be removed from event list.
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ *
+ * kernelENTER_CRITICAL();
+ * {
+ *     if( listLIST_IS_EMPTY( pxEventList ) == pdFALSE )
+ *     {
+ *         xReturn = prvTaskRemoveFromEventList( pxEventList );
+ *     }
+ *     else
+ *     {
+ *         xReturn = pdFALSE;
+ *     }
+ * }
+ * kernelEXIT_CRITICAL();
+ * @endcode
+ * ( listLIST_IS_EMPTY( pxEventList ) == pdFALSE ) is true.
+ */
+void test_granular_locks_tasks_xTaskRemoveFromEventList_list_is_not_empty( void )
+{
+    TCB_t xTaskTCB = { NULL };
+    List_t xEventList;
+    BaseType_t xReturn;
+
+    vListInitialise( &xEventList );
+    xTaskTCB.xEventListItem.pvOwner = &xTaskTCB;
+    xTaskTCB.xStateListItem.pvOwner = &xTaskTCB;
+    listINSERT_END( &xEventList, &xTaskTCB.xEventListItem );
+    listINSERT_END( &pxReadyTasksLists[ 0 ], &xTaskTCB.xStateListItem );
+    pxCurrentTCBs[ 0 ] = &xTaskTCB;
+    uxSchedulerSuspended = 0U;
+    xSchedulerRunning = pdTRUE;
+
+    vFakePortDisableInterrupts_ExpectAndReturn( 0 );
+    vFakePortGetSpinlock_Expect( 0, &xTaskSpinlock );
+    vFakePortGetSpinlock_Expect( 0, &xISRSpinlock );
+
+    vFakePortReleaseSpinlock_Expect( 0, &xISRSpinlock );
+    vFakePortReleaseSpinlock_Expect( 0, &xTaskSpinlock );
+    vFakePortEnableInterrupts_Expect();
+
+    xReturn = xTaskRemoveFromEventList( &xEventList );
+
+    TEST_ASSERT_EQUAL( pdFALSE, xReturn );
+    TEST_ASSERT_EQUAL( 0, xEventList.uxNumberOfItems );
+}
+
+/**
+ * @brief xTaskRemoveFromEventList - Remove a task from event list.
+ *
+ * The event list is empty. xReturn is pdFALSE.
+ *
+ * <b>Coverage</b>
+ * @code{c}
+ *
+ * kernelENTER_CRITICAL();
+ * {
+ *     if( listLIST_IS_EMPTY( pxEventList ) == pdFALSE )
+ *     {
+ *         xReturn = prvTaskRemoveFromEventList( pxEventList );
+ *     }
+ *     else
+ *     {
+ *         xReturn = pdFALSE;
+ *     }
+ * }
+ * kernelEXIT_CRITICAL();
+ * @endcode
+ * ( listLIST_IS_EMPTY( pxEventList ) == pdFALSE ) is false.
+ */
+void test_granular_locks_tasks_xTaskRemoveFromEventList_list_is_empty( void )
+{
+    TCB_t xTaskTCB = { NULL };
+    List_t xEventList;
+    BaseType_t xReturn;
+
+    vListInitialise( &xEventList );
+    xTaskTCB.xEventListItem.pvOwner = &xTaskTCB;
+    xSchedulerRunning = pdTRUE;
+    pxCurrentTCBs[ 0 ] = &xTaskTCB;
+
+    vFakePortDisableInterrupts_ExpectAndReturn( 0 );
+    vFakePortGetSpinlock_Expect( 0, &xTaskSpinlock );
+    vFakePortGetSpinlock_Expect( 0, &xISRSpinlock );
+
+    vFakePortReleaseSpinlock_Expect( 0, &xISRSpinlock );
+    vFakePortReleaseSpinlock_Expect( 0, &xTaskSpinlock );
+    vFakePortEnableInterrupts_Expect();
+
+    xReturn = xTaskRemoveFromEventList( &xEventList );
+
+    TEST_ASSERT_EQUAL( pdFALSE, xReturn );
```

Comment:
```suggestion
    TEST_ASSERT_EQUAL( pdFALSE, xReturn );
    TEST_ASSERT_EQUAL( 0, xEventList.uxNumberOfItems );
```
To align the behavior with `test_granular_locks_tasks_xTaskRemoveFromEventListFromISR_list_is_empty`.

Reply:
This is list is empty case. How about we check the list item status instead?